### ISSUE TITLE
broker was died on netty's pipeline.remove() invoke?

### ISSUE THREAD

**Issue by tixo** [moquette-io/moquette#102]
I use jmeter do a load test . when I use 600 threads and 1 seconds to run , a few later , the broker was died,  I just change the m_disruptor's bufferSize and m_executor's name in SimpleMessaging class, so the code is

```
public void init(IConfig configProps) {
        m_executor = Executors.newSingleThreadExecutor(new DefaultThreadFactory("MessagingDispatcher"));
        m_disruptor = new Disruptor<>(ValueEvent.EVENT_FACTORY, 128, m_executor);
...
         }
```

the thread dump:

MessagingDispatcher-1-1" prio=6 tid=0x000000000b5e7800 nid=0x2350 in Object.wait() [0x000000000c34e000]
   java.lang.Thread.State: WAITING (on object monitor)
    at java.lang.Object.wait(Native Method)
    - waiting on <0x00000000ff878490> (a io.netty.util.concurrent.PromiseTask)
    at java.lang.Object.wait(Object.java:503)
    at io.netty.util.concurrent.DefaultPromise.await(DefaultPromise.java:254)
    - locked <0x00000000ff878490> (a io.netty.util.concurrent.PromiseTask)
    at io.netty.util.concurrent.DefaultPromise.await(DefaultPromise.java:32)
    at io.netty.util.concurrent.AbstractFuture.get(AbstractFuture.java:31)
    at io.netty.channel.DefaultChannelPipeline.waitForFuture(DefaultChannelPipeline.java:552)
    at io.netty.channel.DefaultChannelPipeline.remove(DefaultChannelPipeline.java:340)
    at io.netty.channel.DefaultChannelPipeline.remove(DefaultChannelPipeline.java:305)
    at org.eclipse.moquette.server.netty.NettyChannel.setIdleTime(NettyChannel.java:60)
    at org.eclipse.moquette.spi.impl.ProtocolProcessor.processConnect(ProtocolProcessor.java:229)
    at sun.reflect.GeneratedMethodAccessor29.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at org.eclipse.moquette.spi.impl.AnnotationSupport.dispatch(AnnotationSupport.java:64)
    at org.eclipse.moquette.spi.impl.SimpleMessaging.onEvent(SimpleMessaging.java:177)
    at org.eclipse.moquette.spi.impl.SimpleMessaging.onEvent(SimpleMessaging.java:67)
    at com.lmax.disruptor.BatchEventProcessor.run(BatchEventProcessor.java:128)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:137)
    at java.lang.Thread.run(Thread.java:744)

---

when I remove the line of session.setIdleTime()  in ProtocolProcessor class, the broker work well

```
void processConnect(ServerChannel session, ConnectMessage msg) {
...
session.setIdleTime(Math.round(keepAlive * 1.5f)); //remove this
...
```

**Comment by andsel** [comment]
Hi @tixo I can't understand why you reduced so much the RingBuffer slots (that could be the cause) and how the change in the name use by Thread factory to "MessagingDispatcher" impact the problem, it's completely rationally unbounded (the name of the thread pool with the wait or a die in the broker)

**Comment by tixo** [comment]
Hi @andsel ,  sometimes,  moquette was died in my production env .  so in order to reappear the problem rapid , I set a small RingBuffer to run with Jmeter. and set a name to SimpleMessage ' s executor for look up thread dump

**Comment by andsel** [comment]
Hi @tixo thanks for your hint, now I'm able to reproduce and investigate the bug (I've lowered the size f ring buffer to 32, and its manifest with only 40 subscribers and 40 publishers).
So what I can suppose (then I'll see how to fix) is that ProtocolProcessor  process income events from the ring slower than could be inserted, these makes the ring full and this let the threads used by Netty to be parked trying to insert data into the ring. On the other side of the ring, the single thread of ProtocolProcessor try to set the idle timeout for the CONNECT message it is processing, but this change (the manipulation of the income pipeline) has to be executed by the Netty's  thread that is blocked on the push side of ring. 
 I'll find some thing to fix it, thank's a lot for the issue

**Comment by andsel** [comment]
Hi @tixo I changed the way Moquette works so no more a ring buffer and single thread processor,this means that latency errors shouldn't happened and this bug is fixed.
